### PR TITLE
adds/updates jsdoc comments, expose useK8sWatchResource(s) hooks from SDK and adds typing

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes:**

https://issues.redhat.com/browse/HAC-391

**Description:**

- [x] adds/updates jsdoc comments
- [x] expose useK8sWatchResource(s) hooks from SDK
- [x] adds typing

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts [hunk 1]
```diff
@@ -229,6 +303,15 @@ export const k8sKill = <R extends K8sResourceCommon>(
...
 };
 
+type OptionsKill<R> = BaseOptions & {
+  model: K8sModel;
+  resource: R;
+  requestInit: RequestInit;
+  json: Record<string, any>;
+};
+
+type K8sKillResource = <R extends K8sResourceCommon>(options: OptionsKill<R>) => Promise<Response>;
```
Comment (andrewballantyne) [inline frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts:313]:
```suggestion
type OptionsDelete<R> = BaseOptions & {
  model: K8sModel;
  resource: R;
  requestInit: RequestInit;
  json: Record<string, any>;
};

type K8sDeleteResource = <R extends K8sResourceCommon>(options: OptionsDelete<R>) => Promise<Response>;
```
We need to pick a side of the fence :)

File: frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts [hunk 2]
```diff
@@ -298,4 +390,9 @@ export const k8sList = (
  * See more {@link https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html}
  * @return A promise that resolves to the response
  * * */
-export const k8sListResource = adapterFunc(k8sList, ['model', 'queryParams', 'raw', 'requestInit']);
+export const k8sListResource: K8sListResource = adapterFunc(k8sList, [
+  'model',
+  'queryParams',
+  'raw',
```
Comment (andrewballantyne) [inline frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts:396]: Still a little odd that you're not specifying `raw` in the JSDoc

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts (hunk 1 above)
Comment (invincibleJai) [reply frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts]: sorry missed, updated :)

File: frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts (hunk 2 above)
Comment (invincibleJai) [reply frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts:398]: yeah, this is not being used across so in initial PR the thought was not to expose it in a new signature.

https://github.com/openshift/console/pull/10243#discussion_r732973059

image

Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts:398]: My only concern is that this is something that is almost self documented in a very limited way due to the signature of the `adapterFunc` and thus looks like a gap in the JSDoc. But given Sam believes there is only 1 impacted location, I suppose we could look to move away from it... not sure how we deprecate a single param though.

**Review by andrewballantyne — APPROVED** [review]
Looking good, thanks for fixing up and closing this gap @invincibleJai 

My only follow up point here is I'm still not comfortable with the way the `adapterFunc` signature looks in respect to the way JSDoc -- but that's a discussion we have already have had and I guess this will be okay for now.

**Comment by andrewballantyne** [comment]
No customer-facing documentation needed, this is all internal to the SDK.
/label docs-approved
/label px-approved

No QE needed, this is a documentation update PR.
/label qe-approved

(just in case my previous approval doesn't add this)
/lgtm

/retest

**Pull request merged**